purchases/src/main/kotlin/com/revenuecat/purchases/Offerings.kt

```diff
@@ -58,9 +69,13 @@ internal data class Placements(
     val offeringIdsByPlacement: Map<String, String?>,
 )
 
-private fun Offering.withPlacement(placementId: String): Offering {
+internal fun Offering.withPresentedContext(placementId: String?, targeting: Targeting?): Offering {
```

Comment:
You could potentially pass the current values as default values for these parameters I think? That way we would avoid overwriting the wrong value accidentally.